Modules/Synchronization/src/Synchronization.Application/Datawallets/Queries/GetModifications/Handler.cs

```diff
@@ -50,49 +42,15 @@ private async Task<List<DatawalletModificationDTO>> MapToDtos(IEnumerable<Datawa
     {
         var datawalletModifications = modifications as DatawalletModification[] ?? modifications.ToArray();
 
-        var blobReferences = datawalletModifications.Where(m => !m.BlobReference.IsNullOrEmpty()).Select(m => m.BlobReference).Distinct();
-        var blobs = await Task.WhenAll(blobReferences.Select(r =>
-        {
-            try
-            {
-                return _blobStorage.FindAsync(_blobOptions.RootFolder, r);
-            }
-            catch (NotFoundException)
-            {
-                throw new Exception($"Blob with reference '{r}' not found.");
-            }
-        }));
-
-        var payloads = blobs
-            .Select(b => JsonSerializer.Deserialize<Dictionary<long, byte[]>>(b))
-            .SelectMany(b => b)
-            .ToDictionary(b => b.Key, b => b.Value);
-
-        var mappingTasks = datawalletModifications.Select(m => MapToDto(m, payloads));
+        var mappingTasks = datawalletModifications.Select(MapToDto);
 
         return (await Task.WhenAll(mappingTasks)).ToList();
     }
 
-    private async Task<DatawalletModificationDTO> MapToDto(DatawalletModification modification, Dictionary<long, byte[]> payloads)
+    private async Task<DatawalletModificationDTO> MapToDto(DatawalletModification modification)
     {
         var dto = _mapper.Map<DatawalletModificationDTO>(modification);
-
-        if (modification.BlobReference.IsNullOrEmpty())
-        {
-            try
-            {
-                dto.EncryptedPayload = await _blobStorage.FindAsync(_blobOptions.RootFolder, modification.Id);
-            }
-            catch (NotFoundException)
-            {
-                // blob not found means that there is no payload for this modification
-            }
-        }
-        else
-        {
-            payloads.TryGetValue(modification.Index, out var payload);
-            dto.EncryptedPayload = payload;
-        }
+        dto.EncryptedPayload = modification.EncryptedPayload;
```

Comment:
This line can be removed, since it's done automatically by AutoMapper based on property name convention.